##  ДЗ Урок 1

### Задача 1

Группа студентов изучает 10 различных дисциплин. Сколькими способами можно составить расписание на понедельник, если в этот день должно быть 4 разных занятия?

In [1]:
import numpy as np

Так как нам не важен порядок выставления дисциплин воспользуемся формулой комбинаторики (сочетания). Напишем функцию для расчета

In [2]:
def combinations (n, k):
    return np.math.factorial(n) // (np.math.factorial(k) * np.math.factorial(n - k))

n = 10 (сколько всего дисциплин изучает группа); <br>
k = 4 (сколько дисциплин нужно поставить на понедельник)

In [3]:
combinations(10, 4)

210

Ответ: существует 210 способов составить расписание на понедельник

### Задача 2

Из колоды в 52 карты вынимают случайным образом 4 карты. Найти число исходов, соответствующих тому, что был вытянут хотя бы один туз.

Условно разделим колоду на две части, 4 туза и все остальные карты в количестве 48 шт. Так как нам не важен порядок вытягивания карт, воспользуемся формулой комбинаторики (сочетания). Будем использовать функцию из первого задания

Рассчитаем число сочетаний для вытягивания туза <br>
n = 4 (кол-во тузов в колоде) <br>
k = 1 (сколько тузов необходимо вытащить) 

In [4]:
combinations(4, 1)

4

Рассчитаем количество сочетаний вытягивания оставшихся трех карт <br>
n = 48 (оставшееся количество карт, без тузов)<br>
k = 3 (сколько карт необходимо вытащить)

In [5]:
combinations(48, 3)

17296

Так как нам необходимо вытащить из колоды один туз И еще три карты, то результаты сочетаний умножаются

In [6]:
result = combinations(4, 1) * combinations(48, 3)
print(result)

69184


### Задача 3
Семь человек рассаживаются наудачу на скамейке. Какова вероятность того, что два определённых человека будут сидеть рядом?

Чтобы найти вероятность найдем количество исходов, когда два человека будут сидеть рядом и разделим на общее количество исходов

В данном случае нам важен порядок, поэтому воспользуемся формулой комбинаторики (размещения). Напишем функцию для рассчета

In [9]:
def arrangements(n, k):
    return np.math.factorial(n) // np.math.factorial(n - k)

n = 7 (количество человек) <br>
k = 2 (количество человек, которые должны сесть рядом)

In [18]:
arrangements(7, 2)

42

Общее количество исходов найдем с помощью формулы комбинаторики (перестановки). Напишем функцию для рассчета

In [11]:
def permutations(n):
    return np.math.factorial(n)

In [12]:
permutations(7)

5040

In [13]:
result = arrangements(7, 2) / permutations(7)
print(result)

0.008333333333333333


Ответ: Вероятность того, что два определенных человека сядут рядом 0.8%

### Задача 4
Из 60 вопросов, входящих в экзаменационные билеты, студент знает 50. Какова вероятность того, что среди трёх наугад выбранных вопросов студент знает: а) все? б) два?

Для решения задачи воспользуемся формулой полной вероятности (получился странный ответ, прокомментируйте пожалуйста можно ли решить эту задачу таким способом)<br>
$$P(A) = P(H_1) \cdot P(A | H_1) + P(H_2) \cdot P(A | H_2) + \dots + P(H_n) \cdot P(A | H_n)$$ <br>
Где $$P(H_1) = \frac{1}{60}$$
$$P(A | H_1) = \frac{50}{60}$$
$$P(H_2) = \frac{1}{59}$$
$$P(A | H_2) = \frac{49}{59}$$
$$P(H_3) = \frac{1}{58}$$
$$P(A | H_3) = \frac{48}{58}$$

In [16]:
p = ((1 / 60) * (50 / 60)) + ((1 / 59) * (49 / 59)) + ((1 / 58) * (48 / 58))
print(p)

0.04223403141732848


Попробуем решить задачу через деление интересующих нас исходов на общее количество исходов

n = 50 (сколько вопросов знает студент) <br>
k = 3 (сколько билетов нужно вытянуть)

In [19]:
combinations(50, 3)

19600

Общее количество исходов <br>
n = 60 (сколько всего вопросов) <br>
k = 3 (сколько билетов нужно вытянуть)

In [20]:
combinations(60, 3)

34220

In [21]:
result = combinations(50, 3) / combinations(60, 3)
print(result)

0.5727644652250146


Вероятность того, что 3 из 3 вопросов будут которые студент знает равна 57% 

Рассчитаем второй случай, вероятность, что из 3 вопросов студент будет знать 2

n = 50 (сколько вопросов знает студент) <br>
k = 2 (сколько билетов нужно вытянуть)

In [22]:
combinations(50, 2)

1225

In [23]:
combinations(10, 1)

10

Количество исходов, когда студент выберет 2 вопроса которые знает и 1 который не знает равен произведению дву исходов рассчитаных выше

In [25]:
b = combinations(50, 2) * combinations(10, 1)
print(b)

12250


In [26]:
result = b / combinations(60, 3)
print(result)

0.35797779076563413


Вероятность того, что студент выберет 2 вопроса которые знает и 1 который не знает равна 36%

### Задача 5

Бросается игральная кость. Пусть событие A - появление чётного числа, событие B - появление числа больше трёх. Являются ли эти события независимыми?

События являются независимыми, так как вероятность события А не зависит от вероятности события В и наоборот, вероятность события В не зависит от вероятности осбытия А.